arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -0,0 +1,644 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "init_cameras_landmarks.h"
+
+#include <kwiversys/CommandLineArguments.hxx>
+#include <kwiversys/SystemTools.hxx>
+
+#include <vital/algo/initialize_cameras_landmarks.h>
+#include <vital/algo/video_input.h>
+#include <vital/config/config_block_io.h>
+#include <vital/config/config_block.h>
+#include <vital/config/config_parser.h>
+#include <vital/io/camera_from_metadata.h>
+#include <vital/io/camera_io.h>
+#include <vital/io/landmark_map_io.h>
+#include <vital/io/metadata_io.h>
+#include <vital/io/track_set_io.h>
+#include <vital/plugin_loader/plugin_manager.h>
+#include <vital/util/get_paths.h>
+
+#include <fstream>
+#include <iostream>
+
+namespace kwiver {
+namespace arrows {
+namespace mvg {
+
+using kwiver::vital::feature_track_set_sptr;
+using kwiver::vital::algo::initialize_cameras_landmarks;
+using kwiver::vital::algo::initialize_cameras_landmarks_sptr;
+using kwiver::vital::algo::video_input;
+using kwiver::vital::algo::video_input_sptr;
+using kwiver::vital::camera_map_sptr;
+using kwiver::vital::camera_perspective;
+using kwiver::vital::camera_sptr;
+using kwiver::vital::landmark_map_sptr;
+using kwiver::vital::sfm_constraints;
+using kwiver::vital::sfm_constraints_sptr;
+
+
+
+
+namespace {
+
+typedef kwiversys::SystemTools ST;
+typedef kwiversys::CommandLineArguments argT;
+
+kwiver::vital::logger_handle_t main_logger( kwiver::vital::get_logger( "init_cameras_landmarks" ) );
+
+//-------------------------------------------------------------------
+std::string
+frameName(kwiver::vital::frame_id_t frame,
```

Comment:
We should probably just move this helper function to the same header and source file that contains `basename_from_metadata` and just make this a version of `basename_from_metadata` that takes a `metadata_vector`.